src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -1,13 +1,94 @@
 import React from 'react';
-import {getFollowing} from '../../../service/profile';
+import {Context} from '../../../context';
+import DiscoveryAction from '../../../context/actions/discoveryAction';
+import following from '../../../context/actions/following';
 import {getFollowedDomain} from '../../../service/domain';
+import {getFollowing} from '../../../service/profile';
 import {getFollowingTopic} from '../../../service/topics';
-import following from '../../../context/actions/following';
-import {Context} from '../../../context';
 
 const useDiscovery = () => {
   const [, followingDispatch] = React.useContext(Context).following;
   const [refreshing, setRefreshing] = React.useState(false);
+  const [discovery, discoveryDispatch] = React.useContext(Context).discovery;
+
+  const users = React.useMemo(() => {
+    return discovery.initialUsers.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [discovery]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `users` variable is memoized here to optimize performance. However, the dependency array only includes `discovery`, which means that this useMemo will re-run whenever any property of `discovery` changes. If there are properties of `discovery` that can change without affecting `users`, this could lead to unnecessary re-renders. Consider specifying the exact dependencies that affect `users`.

```diff
-  }, [discovery]);
+  }, [discovery.initialUsers]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->